instrumentation/opentelemetry_dataloader/lib/opentelemetry_dataloader.ex

```diff
@@ -0,0 +1,99 @@
+defmodule OpentelemetryDataloader do
+  @moduledoc """
+  Module for automatic instrumentation of Dataloader.
+
+  It works by listening to `[:dataloader, :source, :run/:batch, :start/:stop]` telemetry events.
+  """
+
+  @tracer_id __MODULE__
+
+  @run_start [:dataloader, :source, :run, :start]
+  @run_stop [:dataloader, :source, :run, :stop]
+
+  @batch_start [:dataloader, :source, :batch, :run, :start]
+  @batch_stop [:dataloader, :source, :batch, :run, :stop]
+
+  @doc """
+  Initializes and configures the telemetry handlers.
+  """
+  def setup(_opts \\ []) do
+    config = %{
+      tracer_id: @tracer_id
+    }
+
+    :telemetry.attach_many(
+      {__MODULE__, :run},
+      [
+        @run_start,
+        @run_stop
+      ],
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    :telemetry.attach_many(
+      {__MODULE__, :batch},
+      [
+        @batch_start,
+        @batch_stop
+      ],
+      &__MODULE__.handle_event/4,
+      config
+    )
+  end
+
+  def handle_event(event, measurements, metadata, config)
+
+  def handle_event(@run_start, _measurements, metadata, config) do
+    parent_ctx = OpentelemetryProcessPropagator.fetch_parent_ctx(4, :"$callers")
+
+    if parent_ctx != :undefined do
+      OpenTelemetry.Ctx.attach(parent_ctx)
+    end
+
+    OpentelemetryTelemetry.start_telemetry_span(
+      config.tracer_id,
+      "dataloader.run",
+      metadata,
+      %{
+        kind: :client
+      }
+    )
+  end
+
+  def handle_event(@run_stop, _measurements, metadata, config) do
+    OpentelemetryTelemetry.set_current_telemetry_span(config.tracer_id, metadata)
```

Comment:
I don't think you need to do this.

Reply:
I got that from here: https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/fb5598800569cd45d97a891b001f4b39efbaf7d7/instrumentation/opentelemetry_absinthe/lib/opentelemetry_absinthe/execute_handler.ex#L51 but I'l remove it. 

Should it also be removed in the @batch_stop handle_event?

Reply:
Oh, and now I see its also in Phoenix https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex#L136

@bryannaegele you'll have to explain this one to me :)

Reply:
@tsloughter it's possible another span is open, especially with Phoenix due to when controller functions return, so it was a safeguard to ensure the Phoenix span was closed properly to not lose the trace. I don't know that this is as much of a concern in other libraries, so probably not needed, but also isn't harmful if someone used it.

The other use case would be if some process was creating many spans that are getting closed asynchronously and you therefore have more than one in progress. An example would be a genserver.

I can't speak to the dataloader library, but those are the cases when I'd employ that function.

Reply:
Ok, yea, I'm guessing I thought `OpentelemetryTelemetry` worked differently when I originally commented.